src/vanilla/atom.ts

```diff
@@ -75,6 +75,11 @@ export function atom<Value, Args extends unknown[], Result>(
 // read-only derived atom
 export function atom<Value>(read: Read<Value>): Atom<Value>
 
+// Value with default as initial value
+export function atom<Value = undefined>(
+  read: Read<Value | undefined>,
+): Atom<Value | undefined>
```

Comment:
Hmm, I was expecting something like this:
```suggestion
export function atom<Value>(): PrimitiveAtom<Value | undefined>
```

Reply:
This works but, in `/src/vanilla/atom.ts`, I get
```console
This overload signature is not compatible with its implementation signature.ts(2394)
atom.ts(92, 17): The implementation signature is declared here.
```
The signature (parameter) needs to be added.

Reply:
Yeah, let's do that. Just make it optional?

Reply:
Done